app/models/header_lookup.rb

```diff
@@ -61,24 +63,32 @@ def pid_lookup_by_field(term="", field="")
         end
       end
 
-      memoized_terms[term] = term_pid
-      File.write(memoized_file, memoized_terms)
+      @@all_memoized_terms[memoized_vocab][term] = term_pid
+
+      File.write(memoized_file, @@all_memoized_terms[memoized_vocab].to_json)
```

Comment:
If we want to edit these files by hand/see diffs better, something like:
https://docs.ruby-lang.org/en/2.7.0/JSON.html#method-i-pretty_generate is probably what we want (otherwise we will get a very long 1 line json object). I leave it as your call.

Reply:
This makes more sense. I actually have never used that method so I'm going to test it a bit to make sure it's not going to do anything unexpected.